ecosystem/manager.py

```diff
@@ -52,6 +52,63 @@ def recompile(self):
         """Recompile `members.json` from human-readable files."""
         self.dao.compile_json()
 
+    def build_website(self):
+        """Generates the ecosystem web page reading `members.json`."""
+        environment = Environment(loader=FileSystemLoader("ecosystem/html/"))
+        website_template = environment.get_template("webpage.html")
+        card_template = environment.get_template("card.html")
+        tag_template = environment.get_template("tag.html")
+        link_template = environment.get_template("link.html")
+
+        sections = {
+            "provider": "",
+            "applications": "",
+            "other": "",
+        }
+
+        projects = DAO("ecosystem/resources/").storage.read()
+
+        count_read_more = 1
+        max_chars_description = 500
+        for _, repo in projects.items():
+            # Card tags
+            tags = ""
+            for label in repo.labels:
+                tags += tag_template.render(color="purple", title=label, text=label)
+
+            # Card links
+            links = link_template.render(url=repo.url, place="repository")
+            if repo.website:
+                links += link_template.render(url=repo.website, place="website")
+
+            # Card description
+            if len(repo.description) - max_chars_description >= 0:
+                description = [repo.description[:400], repo.description[400:]]
+                id_read_more = str(count_read_more)
+                count_read_more += 1
+            else:
+                description = [repo.description, ""]
+                id_read_more = "None"
+
+            # Create the card
+            card = card_template.render(
+                title=repo.name,
+                tags=tags,
+                description_visible=description[0],
+                description_hidden=description[1],
+                id_read_more=id_read_more,
+                links=links,
+            )
+
+            # Adding the card to a section
+            sections[repo.group] += card
+
+        return website_template.render(
+            section1_cards=sections["provider"],
+            section2_cards=sections["applications"],
+            section3_cards=sections["other"],
```

Comment:
Rather than `section1_cards`, I think it's better to be explicit with named sections like `section_provider_cards`. We otherwise risk mixing up what is what.